internal/testing/check_mod_tidy.sh

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+# Copyright 2019 The Go Cloud Development Kit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script checks to see if `go mod tidy` has been run on a module.
+#
+# With no arguments, it checks the module in the current directory.
+# Alternatively, it accepts a single argument to check the module in the given
+# directory.
+#
+# It exits with status 1 if "go mod tidy && go list -deps ./..." would
+# make changes.
+#
+# TODO(rvangent): Replace this with `go mod tidy --check` when it exists:
+# https://github.com/golang/go/issues/27005.
+#
+# TODO(rvangent): Drop the "go list" part here and in gomodcleanup.sh once
+# https://github.com/golang/go/issues/31248 is fixed.
+
+set -euo pipefail
+
+# If an argument is given, cd there.
+if [[ $# -eq 1 ]]; then
+  cd "$1"
+fi
+
+TMP_GOMOD=$(mktemp)
+TMP_GOSUM=$(mktemp)
+
+function cleanup() {
+  # Restore the original files in case "go mod tidy" made changes.
+  if [[ -f "$TMP_GOMOD" ]]; then
+    mv "$TMP_GOMOD" ./go.mod
+  fi
+  if [[ -f "$TMP_GOSUM" ]]; then
+    mv "$TMP_GOSUM" ./go.sum
+  fi
+}
+trap cleanup EXIT
+
+# Make copies of the current files.
+cp ./go.mod "$TMP_GOMOD"
+cp ./go.sum "$TMP_GOSUM"
+
+# Modifies the files in-place.
+go mod tidy
+go list -deps ./... &> /dev/null || :
```

Comment:
Nit: can you add a comment to explain the `|| :`? This is super cryptic

Reply:
Actually, it's not needed anymore. `go list -deps` fails for some modules, but adding the `./...` makes it always work, and this `|| :` is to make it not fail.

Removed.